tests/unittest.cpp

```diff
@@ -4,6 +4,11 @@
 #include <gtest/gtest.h>
 #endif
 
+// Necessary for proper detection of *BSD
+#if !defined(_MSC_VER)
+#include <sys/param.h>
+#endif
```

Comment:
@GreyCat I suspect this isn't actually necessary since you're only using the `__FreeBSD__` and `__NetBSD__` macros, not the `BSD` macro which indeed requires `#include <sys/param.h>`: https://sourceforge.net/p/predef/wiki/OperatingSystems/#bsd-environment

I think this unnecessary include should be removed. For now, I guess this condition is fine, because the test exclusions should be minimal and these are the only BSD variants that we know are problematic:

https://github.com/kaitai-io/kaitai_struct_cpp_stl_runtime/blob/7b4c29a16b63d3c7738c124423a0087e97ad7834/tests/unittest.cpp#L383

Reply:
I recall in my testing that it actually required that include, but maybe I've indeed got something crossed, so if you have an idea what we can eliminate/optimize here — absolutely feel free to go ahead and change :)

Reply:
I think (at least I haven't seen any single reference on the internet that would suggest otherwise) that `sys/param.h` only defines `BSD` (see [FreeBSD's `sys/param.h`](https://github.com/freebsd/freebsd-src/blob/35e941ce5520b8ef710bb388bd63460aa0ce63eb/sys/sys/param.h#L42), [NetBSD's `sys/param.h`](https://github.com/NetBSD/src/blob/5680363e133011c306e0a21efff4e74c17f9fd4c/sys/sys/param.h#L49)), whereas `__FreeBSD__` and `__NetBSD__` are predefined macros by the compiler, so you don't need to include anything to use them.